src/Data/Tree/AVL/Internal.hs

```diff
@@ -264,69 +292,88 @@ type Params h k v =
 type Base h k v m =
     ( Params h k v
     , MonadCatch m
-    , MonadIO m
     )
 
 -- | Ability to write into the storage.
 type Stores h k v m =
     ( Base h k v m
-    , KVStore h (Isolated h k v) m
+    , KVStore h (IsolatedTemplate Int h k v) m
     )
 
 -- | Ability to read from the storage.
 type Retrieves h k v m =
     ( Base h k v m
-    , KVRetrieve h (Isolated h k v) m
+    , KVRetrieve h (IsolatedTemplate Int h k v) m
     )
 
 -------------------------------------------------------------------------------
 -- * Methods
 -------------------------------------------------------------------------------
 
+mapTilt :: (t1 -> t2) -> MapTemplate t1 h k v -> MapTemplate t2 h k v
+mapTilt f = go
+  where
+    go = \case
+        Free layer -> Free (withTilt f go layer)
+        Pure h     -> Pure  h
+
+withTilt :: (t1 -> t2) -> (s1 -> s2) -> MapLayerTemplate t1 h k v s1 -> MapLayerTemplate t2 h k v s2
+withTilt f go
+    split@ MLBranch
+        { _mlTilt  = t
+        , _mlLeft  = left
+        , _mlRight = right
+        }
+  = split
+        { _mlTilt  = f t
+        , _mlLeft  = go left
+        , _mlRight = go right
+        }
+withTilt _ _ (MLLeaf  h k v) = MLLeaf h k v
+withTilt _ _ (MLEmpty h)     = MLEmpty h
+
+-- | Replaces `Tilt` with `Int` inside the tree to ease serialisation.
+beforeSerialise :: Map h k v -> MapTemplate Int h k v
+beforeSerialise = mapTilt fromEnum
+
+-- | Replaces `Int` with `Tilt` inside the tree to ease deserialisation.
+afterDeserialise :: MapTemplate Int h k v -> Map h k v
+afterDeserialise = mapTilt toEnum
+
+-- | Replaces `Tilt` with `Int` inside the tree to ease serialisation.
+beforeSerialiseLayer :: MapLayer h k v h -> MapLayerTemplate Int h k v h
+beforeSerialiseLayer = withTilt fromEnum id
+
+-- | Replaces `Int` with `Tilt` inside the tree to ease deserialisation.
+afterDeserialiseLayer :: MapLayerTemplate Int h k v h -> MapLayer h k v h
+afterDeserialiseLayer = withTilt toEnum id
```

Comment:
1) I really don't think that writing `Serialise` instances for `Tilt` is such an ordeal that we really need all those helpers. Especially given that it seems that all the `*Template` type aliases are needed only for this.
2) Even if we have them, `Tilt` should be translated into `Word8` instead of `Int` (why on earth we would need 32 bits for 5 values)

Reply:
You're right in (2).

I think, `Tilt` is an... look, I can literally replace `Tilt` with `Word8`. This fixes both problems: 1) removes `Tilt` as API part; 2) removes `*Template`s.